### ISSUE TITLE
enable additional compiler checks by default + fix

### ISSUE THREAD

**Issue by tomholub** [FlowCrypt/flowcrypt-ios#1024]
When trying to find fix for this issue I came across 2 useful flags for Swift concurrency - `-Xfrontend -warn-concurrency -Xfrontend -enable-actor-data-race-checks` (https://twitter.com/olebegemann/status/1421144304127463427)
These flags add additional code checks, which are not enabled by default, and they found 55 errors in our current code.

_Originally posted by @sosnovsky in https://github.com/FlowCrypt/flowcrypt-ios/issues/1022#issuecomment-970268297_

After fixing, could we set these as errors instead of warnings? Also in CI.

**Comment by sosnovsky** [comment]
Xcode automatically shows differernt error types as `error` or `warning`, but current warnings will be errors in Swift 6 - `Warnings in Swift 5.5 identify unsafe constructs, will become errors in Swift 6.`

<img width="632" alt="Screenshot 2021-11-16 at 15 42 05" src="https://user-images.githubusercontent.com/1062093/141996294-f9d132a9-74f9-4b75-a2d1-4981a3d8a1c4.png">

**Comment by tomholub** [comment]
Until then, could enable this:

![image](https://user-images.githubusercontent.com/6306961/141999075-14e09d4d-4e32-4ebe-85d6-628770644627.png)

**Comment by ivan-ushakov** [comment]
@tomholub
Can I start with this one or you have something with higher priority?

**Comment by tomholub** [comment]
You can start with this one. We don't have to treat warnings as errors yet, but should look into addressing as many warnings as possible.

You may want to do several smaller PRs so we don't run into merge conflicts.

Also: `-Xfrontend -warn-concurrency -Xfrontend -enable-actor-data-race-checks` - can this be enabled in the IDE for everyone? So that we all see the warnings.

**Comment by ivan-ushakov** [comment]
I found that enabling `-Xfrontend -warn-concurrency` produces a lot of warning about non sendable types. Not sure if we can fix them now because this requires a lot of changes in code.

I will try to fix errors produced by `-Xfrontend -warn-concurrency`.

**Comment by tomholub** [comment]
Not necessarily all at once. @sosnovsky are these sensible warnings to address?

**Comment by sosnovsky** [comment]
One of Swift language developers suggests not to use `-warn-concurrency` flag for now, as some warnings will be automatically fixed in the next Swift version (https://twitter.com/dgregor79/status/1470940523259056129).

So maybe for now we can fix only `-enable-actor-data-race-checks` warnings

**Comment by tomholub** [comment]
sounds good

**Comment by ivan-ushakov** [comment]
> One of Swift language developers suggests not to use `-warn-concurrency` flag for now, as some warnings will be automatically fixed in the next Swift version (https://twitter.com/dgregor79/status/1470940523259056129).
> 
> So maybe for now we can fix only `-enable-actor-data-race-checks` warnings

With this flag it produces a lot of warnings like:

```
warning: data race detected: @MainActor function at FlowCrypt/InboxViewController.swift:377 was not called on the main thread
warning: data race detected: @MainActor function at FlowCrypt/InboxViewController.swift:221 was not called on the main thread
warning: data race detected: @MainActor function at FlowCrypt/InboxViewController.swift:226 was not called on the main thread
```

Such warnings are related to Texture (mostly). 

```
func tableNode(_: ASTableNode, willBeginBatchFetchWith context: ASBatchContext) {
    context.beginBatchFetching()
    handleBeginFetching(context)
}
```

Adding `Task` + `await` could make code a bit verbose. Are we OK with it?

**Comment by sosnovsky** [comment]
I think we can leave this issue for later, as currently these warnings don't affect app performance.

Maybe in the future we'll replace `Texture` with `SwiftUI`, or `Texture` will add support for actors (but it's a small chance, as it's development isn't very active).

@tomholub are you ok with this?

**Comment by tomholub** [comment]
I'll follow your suggesion

**Comment by tomholub** [comment]
Does it mean we're postponing the whole issue?

**Comment by sosnovsky** [comment]
> Does it mean we're postponing the whole issue?

Yes, as currently enabling these flags produces some misleading warnings and will need code updates which can be unnecessary in the future:

`-Xfrontend -warn-concurrency` - will have more gradual adoption of concurrency in the next Swift versions (https://twitter.com/dgregor79/status/1470940523259056129)

`-Xfrontend -enable-actor-data-race-checks` - most warnings occur in `Texture` framework methods, as it implements custom mechanism for threads handling. May be `MainActor` support will be added there or we'll fix these issues by migrating to SwiftUI https://github.com/FlowCrypt/flowcrypt-ios/issues/782

**Comment by tomholub** [comment]
got it. Let's re-evaluate at the end of the year.
